File: desing-pattens-1/src/modelo/implementacao/desconto/SemDesconto.java
package modelo.implementacao.desconto;

import interfaces.IDesconto;
import modelo.Orcamento;

public class SemDesconto implements IDesconto {

	@Override
	public double calcula(Orcamento orcamento) {

		return 0;
	}

	@Override
	public void setProximo(IDesconto desconto) {

	}

}

File: desing-pattens-1/src/main/java/modelo/implementacao/desconto/SemDesconto.kt
package main.java.modelo.implementacao.desconto

import main.java.interfaces.IDesconto
import main.java.modelo.Orcamento

class SemDesconto : IDesconto {

    override fun calcula(orcamento: Orcamento): Double {

        return 0.0
    }

    override fun setProximo(desconto: IDesconto) {

    }

}
